src/main/java/com/xxxx/example/Servlet04.java

```diff
@@ -16,17 +16,19 @@
 public class Servlet04 extends HttpServlet {
     @Override
     protected void service(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        req.setCharacterEncoding("UTF-8");
         resp.setCharacterEncoding("UTF-8");
-        Database myDatabase = new Database("<connection string goes here>", "DatingAppStaging");
-        List<Document> doc2 = myDatabase.find_latest_posts(10);
-         // user this to query for a user.
+        Database myDatabase = new Database("", "DatingAppStaging");
```

Comment:
Can we move this hardcoded connection string to be inside a credential file?

The same goes for `DatingAppStaging`, it should be an argument defined at runtime depending on if we are just testing vs actual production. Using `DatingAppStaging` and `DatingAppProduction` respectively

Reply:
That is extraordinary, appreciate it.